---
title: "Analyse_DE.rmd"
output: html_document
---


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}

library(tximport)

#Locate the directory containing the quantification files, associate files' values to a variable 
varion <-c(1,2,3,4,5,6)
fichier <- c()
for (i in varion) {
  fichier[i] <-paste0("/ifb/data/mydatalocal/salmon_alignment_paired_end/quant_",i,"/quant.sf")
  }

#Associate quantification data to the corresponding sample
names(fichier)<-c("Lib1_31_20_S1","Lib2_31_20_S2","Lib3_31_20_S3","Lib4_31_20_S4","Lib5_31_20_S5","Lib6_31_20_S6")
file_wo3 <- fichier[-3] #Remove Lib3

# Associate salmon quantification data to transcripts ID from Trinity and gene ID 

##Create a vector with transcript's ID from Trinity read in a table  
 
trini <- read.table("/ifb/data/mydatalocal/transfer_trinity/Trinity.fasta.gene_trans_map")

##Create a dataframe with transcript IDs and gene IDs correspondence.
names(trini) <-c("GENEID","TXNAME")

##Arrange the data order (transcript-gene IDs) to run tximport
trini1 <-trini[,c(2,1)]  

## Sum transcripts quantification at a gene-level with tximport

txi <- tximport(fichier,type="salmon",tx2gene=trini1)
head(txi$counts)
head(txi$abundance)
head(txi$length)

txi_wo3<- tximport(file_wo3,type="salmon",tx2gene=trini1)
```

```{r}

#Dataset for DESeq2

## Associate sample names/run (rownames) with conditions (colnames)
samp.name <- c("Lib1_31_20_S1","Lib2_31_20_S2","Lib3_31_20_S3","Lib4_31_20_S4","Lib5_31_20_S5","Lib6_31_20_S6")
samp.type <- factor(rep(c("CTL", "IFN"), each = 3))
samples <- data.frame(run=samp.name,condition=samp.type)
samples_wo3 <- data.frame(run=samp.name[-3],condition=samp.type[-3])

##Combine quantification data and the data frame 
library(DESeq2)
ddsTxi <- DESeqDataSetFromTximport(txi,
                                   colData = samples,
                                   design = ~ condition)

ddsTxi_wo3 <- DESeqDataSetFromTximport(txi_wo3,
                                   colData = samples_wo3,
                                   design = ~ condition)
ddsTxi_wo3
```


```{r}
# Measure DE
dds <- DESeq(ddsTxi)
dds_wo3 <- DESeq(ddsTxi_wo3)
res <- results(dds)
res_wo3 <- results (dds_wo3)
res_wo3
write.table(res_wo3, "/ifb/data/mydatalocal/outputs/output_DE_align/dds_res_wo3", append = FALSE, sep = " ", dec = ".",
             row.names = TRUE, col.names = TRUE)
write.csv(res, "/ifb/data/mydatalocal/outputs/output_DE_align/dds_res.csv", append = FALSE, sep = " ", dec = ".",
            row.names = TRUE, col.names = TRUE)
write.table(res, "/ifb/data/mydatalocal/outputs/output_DE_align/dds_res", append = FALSE, sep = " ", dec = ".",
            row.names = TRUE, col.names = TRUE)
```

```{r}
resultats_DESeq2_full_dataset <- write.table(res)
resultats_DESeq2_full_dataset <- write.table(res_wo3)
##a revoir
```

```{r}
table(res$padj < 0.1)
table(res_wo3$padj < 0.1)
```

```{Graphic lecture}
plotMA(res, ylim=c(-2,2))
plotMA(res_wo3, ylim=c(-2,2))
vsd <- vst(dds, blind=FALSE)
vsd_wo3 <- vst(dds_wo3, blind=FALSE)
plotPCA(vsd, intgroup=c("condition","run"))
plotPCA(vsd_wo3, intgroup=c("condition","run"))
```

 


```{r}
#Instal Library BiomaRt
ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")
tx2geneHomo = biomaRt::getBM(attributes=c('ensembl_transcript_id', 
                                      'ensembl_gene_id', 'external_gene_name'),mart=ensembl)
library(biomaRt)
head(tx2geneHomo)
```

```{r}
#Assess other packages installation
library(dplyr)
library(tidyr)#restart R si pb (aller dans session)
library(data.table)
library(ggplot2) #graphs

```


```{r}
#Import blast alignment datas 
blastHomo<-read.table("/ifb/data/mydatalocal/download/DataBats.blast")
blastHomo<-data.table(blastHomo)
head (blastHomo)
```


```{r}
blastHomo<-blastHomo[, c("trid", "version") := tstrsplit(V2, ".", fixed=TRUE)][]
blastHomo<-blastHomo[, c("gene", "alttr") := tstrsplit(V1, "_i", fixed=TRUE)][]
head (blastHomo)
blastHomo1 <- blastHomo%>% group_by(gene)%>% slice_max(order_by = V12, n = 1, with_ties = TRUE)
head (blastHomo1)
blastHomoName<-merge(tx2geneHomo,blastHomo1,by.x="ensembl_transcript_id",by.y="trid")
blastHomoNameUniq<-unique(blastHomoName[,c("ensembl_gene_id","external_gene_name","gene")])
head(blastHomoNameUniq)
```


```{r}
resUsName=merge(as.data.frame(res),blastHomoNameUniq,by.x=0,by.y="gene")
resUsName

resUsNamewo3=merge(as.data.frame(res_wo3),blastHomoNameUniq,by.x=0,by.y="gene")
resUsNamewo3

View(resUsName)
View(resUsNamewo3)
write.csv(resUsName,file="/ifb/data/mydatalocal/outputs/count_blast_deseq")


write.csv(resUsNamewo3,file="/ifb/data/mydatalocal/outputs/count_blast_deseq")
#pour télécharger le tableau de résultats

```


```{r}
#Classification de resUsName:our table
AllReg <- filter(resUsName, padj<0.05)
AllReg
write.csv(AllReg,file="/ifb/data/mydatalocal/outputs/count_blast_deseq_allreg")

UpReg <- filter(resUsName, padj<0.05 & resUsName$log2FoldChange>0)
View(UpReg)
write.csv(UpReg,file="/ifb/data/mydatalocal/outputs/count_blast_deseq_up")

DownReg <- filter(resUsName, padj<0.05 & resUsName$log2FoldChange<0)
write.csv(DownReg,file="/ifb/data/mydatalocal/outputs/count_blast_deseq_down")


```

```{r}

table(resUsName$padj <0.05)
table(resUsName$padj <0.05 & resUsName$log2FoldChange>0)
table(resUsName$log2FoldChange>0)
table(resUsName$padj <0.05 & resUsName$log2FoldChange<0)
```




```{r}
#Comparison with  Holzer data
## Holzer data importation
resBatHolzer<-read.csv("https://www.rna.uni-jena.de/supplements/mda/data/deseq2/RNA/Mock_6h_RNA_vs_IFN_6h_RNA/deseq2_Mock_6h_RNA_IFN_6h_RNA_full.csv")

##Listing external gene names of DE genes
write.csv(as.data.frame(unique(resUsName$external_gene_name)), 
          file="/ifb/data/mydatalocal/download/DE_align/genename_all.csv",quote=FALSE,row.names = FALSE)

write.csv(as.data.frame(unique(AllReg$external_gene_name)), 
          file="/ifb/data/mydatalocal/download/DE_align/genename_allreg.csv",quote=FALSE,row.names = FALSE)


write.csv(as.data.frame(unique(UpReg$external_gene_name)), 
          file="/ifb/data/mydatalocal/download/DE_align/genename_up.csv",quote=FALSE,row.names = FALSE)
 

write.csv(as.data.frame(unique(DownReg$external_gene_name)), 
          file="/ifb/data/mydatalocal/download/DE_align/genename_down.csv",quote=FALSE,row.names = FALSE)

##Adjust colnames to compare

colnames(resBatHolzer)[2] <- c("external_gene_name")
head(resBatHolzer)

##Merge common genes to Holzer et Us
Holzer_resUs<-merge(resBatHolzer,resUsName,by.x="external_gene_name",by.y="external_gene_name")

##Number of common genes to Holzer& us with interferon induced differential expression or not 

table(them=Holzer_resUs$padj.x<0.05,us=Holzer_resUs$padj.y< 0.05)
#attention à l'ordre dans le merge pour déterminer le x et le y

write.csv(table(them=Holzer_resUs$padj.x<0.05,us=Holzer_resUs$padj.y < 0.05),"/ifb/data/mydatalocal/outputs/output_DE_align/holzer_us.csv")

write.table(table(them=Holzer_resUs$padj.x<0.05,us=Holzer_resUs$padj.y < 0.05), "/ifb/data/mydatalocal/outputs/output_DE_align/holzer_us", quote = TRUE, sep = " ", eol = "\n", na = "NA", dec = ".", row.names = TRUE, col.names = TRUE, qmethod = c("escape", "double"))


```


```{r}
#Statistical significance of Holzer & Us similarities
chisq.test(table(them=Holzer_resUs$padj.x<0.05,us=Holzer_resUs$padj.y<0.05))
#chisq.test(table(us=CommonGenesUsHolze$ourTest,them=CommonGenesUsHolze$theirTest)
#X-squared = 707.46, df = 1, p-value < 2.2e-16
```


```{r}
table(resBatHolzer$padj <0.05)# Number of common genes DE: 1702  VS 12141 NS
table(resBatHolzer$padj <0.05 & resBatHolzer$log2FoldChange>0)# Number of genes significantly UPreg
table(resBatHolzer$padj <0.05 & resBatHolzer$log2FoldChange<0) #Number of genes significatively DOWNregulated 605 VS 16536

```

```{r}
#heatmap
select <- na.omit(resUsName$Row.names[resUsName$padj<0.05 & resUsName$log2FoldChange>0]) #sélection des gènes surexprimés
select <-select[1:10]
row.names(samples)=samples$run

pheatmap(assay(vsd)[select,], cluster_rows=TRUE, show_rownames=TRUE,show_colnames=FALSE,
         cluster_cols=FALSE, annotation_col=samples)
         
         
h=pheatmap(resUsNameVsd[select,10:15], cluster_rows=TRUE, show_rownames=TRUE,show_colnames=FALSE,
           cluster_cols=FALSE, annotation_col=samples, labels_row=resUsNameVsd[select,"external_gene_name"],fontsize_row=5)
ggsave(h,file="mydatalocal/outputs/output_DE_align/h") 

```

```{r}
resUsNameVsd <- merge(resUsName,assay(vsd),by.x="Row.names",by.y=0)
row.names(resUsNameVsd) <- resUsNameVsd$Row.names
cibles <- c("ACTG1","ADAMTSL4")
i=pheatmap(resUsNameVsd[resUsNameVsd$external_gene_name%in%cibles,10:15], cluster_rows=TRUE, show_rownames=TRUE,show_colnames=FALSE,
         cluster_cols=FALSE, annotation_col=samples,labels_row=resUsNameVsd["external_gene_name%in%cibles"],fontsize_row=5)
 
```



